sys/net/link_layer/lorawan/Kconfig

```diff
@@ -0,0 +1,476 @@
+        the different values based on value of port.
+
+choice
+    bool "MAC TX mode"
+    default LORAMAC_DEFAULT_TX_MODE_CNF
+    help
+        Configure the type of uplink packet (confirmed/unconfirmed). A confirmed
+        uplink will expect an acknowledgment from the network server; otherwise,
+        the message will be retransmitted by the number indicated by
+        CONFIG_LORAMAC_DEFAULT_RETX, whereas an unconfirmed message will not
+        expect any acknowledgment back from the network server.
+
+config LORAMAC_DEFAULT_TX_MODE_CNF
+    bool "Confirmable"
+
+config LORAMAC_DEFAULT_TX_MODE_UNCNF
+    bool "Unconfirmable"
+
+endchoice
+
+config LORAMAC_ADVANCED_OPTIONS
+    bool "Enable advanced options"
+
+if LORAMAC_ADVANCED_OPTIONS
+
+config LORAMAC_DEFAULT_RETX
+    int "Uplink retransmissions"
+    default 5
+    range 0 255
+    help
+        Configure the maximum number of retransmissions to be used for a
+        confirmed uplink packet, if no downlink acknowledgment is received from
+        the network.
+
+config LORAMAC_DEFAULT_ADR
+    bool "Enable ADR"
+    help
+        Enable or disable Adaptive Data Rate (ADR). If enabled the end node will
+        inform the network server about the status of ADR using the ADR field in
+        uplink data packet. The network server will then optimize the data rate
+        and the transmission power of the end node based on the information
+        collected from the network.
+
+config LORAMAC_DEFAULT_LINKCHK
+    int "Link check interval (in seconds)"
+    depends on KCONFIG_USEMODULE_RN2XX3
+    default 0
+    help
+        Configure the default link check interval (in seconds). 0 means the link
+        check process is disabled. The link check process is used by an
+        end-device to validate its connectivity with the network. The frame has
+        no payload. When a `LinkCheckReq` is received by the network server via
+        one or multiple gateways, it responds with a `LinkCheckAns` MAC command.
+
+config LORAMAC_DEFAULT_RX1_DELAY
+    int "RX1 window delay (in milliseconds)"
+    default 1000
+    help
+        Configure the RX1 window delay (in milliseconds). This configuration
+        specifies the delay in seconds to open the RX1 window after the end of
+        uplink modulation on the end-node.
+
+config LORAMAC_DEFAULT_AR
+    bool "Enable automatic reply"
+    depends on KCONFIG_USEMODULE_RN2XX3
+    help
+        Enable/disable Automatic Reply (AR) for rn2xx3 module. By enabling the
+        automatic reply, the module will transmit a packet without a payload
+        immediately after a confirmed downlink is received, or when the
+        Frame Pending bit has been set by the server.
+
+choice
+    bool "RX2 window datarate index"
+    default LORAMAC_DEFAULT_RX2_DR_0 if LORAMAC_REGION_EU_868
+    default LORAMAC_DEFAULT_RX2_DR_2 if LORAMAC_REGION_IN_865
+    help
+        Configure the RX2 window datarate. This may be changed only if the
```

Comment:
this one only makes sense if ABP is used. For OTAA it has no effect

Reply:
Yes true. Thanks for pointing that out.